### PR TITLE
add eslint-plugin-react-native-a11y

### PR DESCRIPTION
**Pull request by jpdriver**
## Summary

- this adds [`eslint-plugin-react-native-a11y`](https://github.com/formidablelabs/eslint-plugin-react-native-a11y) to the `eslint-config-react-native-community` package
- as a result, any project using the community config (including the default template) would then additionally get linted for a11y issues
- `eslint-plugin-react-native-a11y` includes several recommended configs. This PR uses the `basic` config which only contains common a11y linting rules that apply across both iOS + Android.
- Additional rules (e.g. checking for `accessibilityIgnoresInvertColors` which is iOS-only) may be optionally enabled by users as with any other ESLint plugin.

## Changelog

[General] [Added] - `eslint-plugin-react-native-a11y` in default ESLint config

## Test Plan

- verified that this works correctly in a new project by including it alongside `eslint-config-react-native-community` (see [installation instructions](https://github.com/FormidableLabs/eslint-plugin-react-native-a11y#configuration))

### Examples
Screen Shot 2021-12-21 at 12 09 03 PM
Screen Shot 2021-12-21 at 12 12 36 PM

> n.b. second example used
> ```
>module.exports = {
>  root: true,
>  extends: ['@react-native-community', 'plugin:react-native-a11y/basic'],
>  rules: {
>    'react-native-a11y/has-valid-accessibility-ignores-invert-colors': 2,
>  },
>};
>```
>this is an additional rule that would not be included with only the `basic` config

### CODE REVIEW

File: packages/eslint-config-react-native-community/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "eslint-plugin-react": "^7.26.1",
     "eslint-plugin-react-hooks": "^4.2.0",
     "eslint-plugin-react-native": "^3.11.0",
+    "eslint-plugin-react-native-a11y": "^3.1.0",
```
Comment (lunaleaps) [inline packages/eslint-config-react-native-community/package.json:26]: I think we'll need to bump the version of this

**Comment by lunaleaps** [comment]
Have we tested if there are any lint errors on `react-native` (not sure if that's what you meant by "new project" since we also lint this codebase with `eslint-config-react-native-community/index.js`

**Comment by jpdriver** [comment]
@lunaleaps running this over the `react-native` codebase itself highlighted a lot of "issues" (seemingly a mix of false positives and genuine positives)

so i made some updates to the plugin in an attempt to fix those, and this is where we are now..

running with `lint --fix` has added `accessible={false}` to any `<Touchable* />` (or `<Pressable />`) components without other props; but i think some of these still aren't right.

there are still false positive cases i don't _think_ the plugin should be adding this for -- so i'll investigate further and update the plugin again.

the plugin is also throwing up warnings for some cases where we have `accessibilityLabel` but no `accessibilityHint`. perhaps that rule should be switched off?

**Comment by jpdriver** [comment]
upon further investigation, having the plugin add `accessible={false}` when you ran `lint --fix` was a bad idea on my part 🤦🏻‍♂️

Although it explicitly highlighted the lack of a11y props to developers, it ultimately harmed the overall experience as it prevented *any* information being communicated by assistive technologies if left unchanged after being added.

So instead I've updated the plugin to add either an `accessibilityRole` or `accessibilityLabel` depending on the context.

I think this approach has advantages over using `defaultProps` because these are clear annotations that hopefully provide explicit yet sensible defaults that devs can then change as needed.

**Pull request closed without merging**